Source/Gauntlet/Assertion/AssertionResult.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  AssertionResult.swift
+//
+//  MIT License
+//
+//  Copyright (c) [2023] The Kroger Co. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in all
+//  copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//  SOFTWARE.
+
+import Foundation
+
+/// The result of an evaluated ``Assertion``.
+public typealias AssertionResult<Value> = Result<Value, FailureReason>
+
+// MARK: - Convenience Constructors
+
+extension AssertionResult where Success == Void {
+    /// A successful result with no value.
+    public static var success: AssertionResult<Void> { .success(()) }
+}
+
+extension AssertionResult {
+    /// Creates an ``AssertionResult`` that is a `failure` with a `message` reason.
+    ///
+    /// - Parameter message: The message describing the failure reason..
+    /// - Returns: An ``AssertionResult`` that is a failure with the specified message.
+    public static func message<T>(_ message: String) -> AssertionResult<T> {
+        .failure(.message(message))
+    }
```

Comment:
I like the convenience functions, but `return .message()` in evaluations doesn't feel like an error occurred at first glance. Could `failure(with message: String)` or `failure(message:)` work here instead?

Likewise, `thrownError()` to `failure(with error: Error)` below so we're not hiding the Result type too much.

Reply:
Updated the names to `failure(message: )` and `failure(thrownError: )`. I want it to explicitly specify that it's a thrown error, as that's how we end up reporting this failure case to XCTest. I don't want the convenience method name to make people think they should ever make and return an error.